overlord/confdbstate/confdbstate.go

```diff
@@ -619,6 +619,69 @@ func GetTransactionForSnapctlGet(ctx *hookstate.Context, view *confdb.View) (*Tr
 	return tx, nil
 }
 
+// LoadConfdbAsync schedules a change to load a confdb, running any appropriate
+// hooks and fulfilling the requests by reading the view and placing the resulting
+// data in the change's data (so it can be read by the client).
+func LoadConfdbAsync(st *state.State, view *confdb.View, requests []string) (changeID string, err error) {
+	account, schemaName := view.Schema().Account, view.Schema().Name
+
+	tx, err := NewTransaction(st, account, schemaName)
+	if err != nil {
+		return "", fmt.Errorf("cannot access confdb view %s/%s/%s: cannot create transaction: %v", account, schemaName, view.Name, err)
+	}
+
+	ts, err := createLoadConfdbTasks(st, tx, view)
+	if err != nil {
+		return "", err
+	}
+
+	chg := st.NewChange("get-confdb", fmt.Sprintf(`Get confdb through "%s/%s/%s"`, account, schemaName, view.Name))
+	if ts != nil {
+		// if there are hooks to run, link the read-confdb task to those tasks
+		clearTxTask, err := ts.Edge(clearTxEdge)
+		if err != nil {
+			return "", err
+		}
+
+		// schedule a task to read the tx after the hook and add the data to the
+		// change so it can be read by the client
+		loadConfdbTask := st.NewTask("load-confdb-change", "Load confdb data into the change")
+		loadConfdbTask.Set("requests", requests)
+		loadConfdbTask.Set("view-name", view.Name)
+
+		loadConfdbTask.Set("tx-task", clearTxTask.ID())
+		loadConfdbTask.WaitFor(clearTxTask)
+		chg.AddAll(ts)
+
+		err = addReadTransaction(st, account, schemaName, clearTxTask.ID())
+		if err != nil {
+			return "", err
+		}
+		chg.AddTask(loadConfdbTask)
+	} else {
+		// no hooks to run so we can just load the values directly into the change
+		// (we still need the change because the API is async)
+		result, err := GetViaView(tx, view, requests)
+		if err != nil {
+			if errors.Is(err, &confdb.NotFoundError{}) {
+				chg.Set("api-data", map[string]interface{}{
+					"confdb-error": err.Error(),
+				})
+				chg.SetStatus(state.DoneStatus)
+				return chg.ID(), nil
+			}
+			return "", fmt.Errorf("cannot read confdb %s/%s: %w", tx.ConfdbAccount, tx.ConfdbName, err)
+		}
+
+		chg.Set("api-data", map[string]interface{}{
+			"confdb-data": result,
+		})
+		chg.SetStatus(state.DoneStatus)
```

Comment:
This could be shared with `doLoadDataIntoChange` to avoid duplication and divergence